pkg/adsc/adsc.go

```diff
@@ -335,23 +336,30 @@ func (a *ADSC) Dial() error {
 }
 
 // Returns a private IP address, or unspecified IP (0.0.0.0) if no IP is available
-func getPrivateIPIfAvailable() net.IP {
+func getPrivateIPIfAvailable() netip.Addr {
 	addrs, _ := net.InterfaceAddrs()
 	for _, addr := range addrs {
-		var ip net.IP
-		switch v := addr.(type) {
-		case *net.IPNet:
-			ip = v.IP
-		case *net.IPAddr:
-			ip = v.IP
-		default:
-			continue
+		var ipAddr netip.Addr
+		addrType := addr.Network()
+		if addrType == "ip+net" {
```

Comment:
Are there some constants for this or are we supposed to use them like this?

Reply:
Hi @ramaraochavali, these constants `ip+net` or `ip` are golang library constants for `addr.Network()`,  please refer to [here](https://cs.opensource.google/go/go/+/master:src/net/ip.go;drc=bd6cd7db07f314443acdb980393f57386d40551f;bpv=1;bpt=1;l=539?q=IPNet&ss=go%2Fgo&gsn=Network&gs=kythe%3A%2F%2Fgo.googlesource.com%2Fgo%3Flang%3Dgo%3Fpath%3Dnet%23method%2520IPNet.Network)for more detail.